src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,565 @@
+
+from ...feature_extraction_utils import BatchFeature, FeatureExtractionMixin
+from ...file_utils import TensorType, is_torch_available
+from ...image_utils import ImageFeatureExtractionMixin, ImageInput, is_torch_tensor
+from ...utils import logging
+
+
+if is_torch_available():
+    import torch
+    from torch import Tensor, nn
+    from torch.nn.functional import interpolate
+
+    if TYPE_CHECKING:
+        from transformers.models.maskformer.modeling_maskformer import MaskFormerForInstanceSegmentationOutput
+
+logger = logging.get_logger(__name__)
+
+
+class MaskFormerFeatureExtractor(FeatureExtractionMixin, ImageFeatureExtractionMixin):
+    r"""
+    Constructs a MaskFormer feature extractor.
+
+    This feature extractor inherits from [`FeatureExtractionMixin`] which contains most of the main methods. Users
+    should refer to this superclass for more information regarding those methods.
+
+    Args:
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Whether to resize the input to a certain `size`.
+        size (`int`, *optional*, defaults to 800):
+            Resize the input to the given size. Only has an effect if `do_resize` is set to `True`. If size is a
+            sequence like `(width, height)`, output size will be matched to this. If size is an int, smaller edge of
+            the image will be matched to this number. i.e, if `height > width`, then image will be rescaled to `(size *
+            height / width, size)`.
+        max_size (`int`, *optional*, defaults to 1333):
+            The largest size an image dimension can have (otherwise it's capped). Only has an effect if `do_resize` is
+            set to `True`.
+        size_divisibility (`int`, *optional*, defaults to 32):
+            Some backbones need images divisible by a certain number, if not passes it detauls to the value used in
+            swin.
+        do_normalize (`bool`, *optional*, defaults to `True`):
+            Whether or not to normalize the input with mean and standard deviation.
+        image_mean (`int`, *optional*, defaults to `[0.485, 0.456, 0.406]`):
+            The sequence of means for each channel, to be used when normalizing images. Defaults to the ImageNet mean.
+        image_std (`int`, *optional*, defaults to `[0.229, 0.224, 0.225]`):
+            The sequence of standard deviations for each channel, to be used when normalizing images. Defaults to the
+            ImageNet std.
+    """
+
+    model_input_names = ["pixel_values", "pixel_mask"]
+
+    def __init__(
+        self,
+        do_resize=True,
+        size=800,
+        max_size=1333,
+        size_divisibility=32,
+        do_normalize=True,
+        image_mean=None,
+        image_std=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.do_resize = do_resize
+        self.size = size
+        self.max_size = max_size
+        self.size_divisibility = size_divisibility
+        self.ignore_label = 255
```

Comment:
I see you set the "ignore_label" here. I'm currently contributing the mIoU metric in the datasets library, where this is called `ignore_index`. I'd argue to use the same name everywhere.

Also, I've an [attribute](https://github.com/huggingface/transformers/blob/074645e32acda6498f16203a8459bb597610f623/src/transformers/models/segformer/feature_extraction_segformer.py#L62-L65) in `SegformerFeatureExtractor` called `reduce_labels`, as some datasets (like ADE20k) require this. Should this be added here as well?

Reply:
I am not 100% this is the case for maskformer as well. I'll check it. Changed `ignore_label` to `ignore_index` and made it a parameter